File: java/leetcode/src/Task328.java
public class Task328 {
    public ListNode oddEvenList(ListNode head) {
        if (head == null) {
            return head;
        }

        ListNode cur = head, prev = head;

        while (cur.next != null && cur.next.next != null) {
            ListNode next = cur.next;
            ListNode nextnext = next.next;
            next.next = nextnext.next;
            nextnext.next = prev.next;
            prev.next = nextnext;

            //next loop
            prev = prev.next;
            cur = next;

        }

        return head;
    }

    public ListNode oddEvenList2(ListNode head) {
        if (head == null) {
            return head;
        }

        ListNode odd = head, even = new ListNode(-1), evenHead = even;

        while (odd.next != null && odd.next.next != null) {
            even.next = odd.next;
            odd.next = odd.next.next;

            odd = odd.next;
            even = even.next;
        }
        even.next = odd.next;
        odd.next = evenHead.next;

        return head;
    }
}

File: py/Task328.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def oddEvenList(self, head: ListNode) -> ListNode:
        if not head:
            return head
        headEven = head.next
        p1, p2 = head, headEven
        while p2 and p2.next:
            p1.next = p2.next
            p1 = p1.next
            p2.next = p1.next
            p2 = p2.next
        p1.next = headEven
        return head
